### PR TITLE
Add better elasticsearch objects

### PR DESCRIPTION
**Pull request by Davidmattei**
Added Request, Response, Document and EMSSource objects/interfaces

Deprecated EMS\CommonBundle\Elasticsearch\Document
Deprecated EMS\CommonBundle\Elasticsearch\DocumentInterface

### CODE REVIEW

File: Elasticsearch/Document/Document.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    public function getContentType(): string
+    {
+        return $this->contentType;
+    }
+
+    public function getEmsId(): string
+    {
+        return "$this->contentType:$this->id";
```
Comment (dameert) [inline Elasticsearch/Document/Document.php]: not a fan of "" variable resolution. readability etcetera

File: Elasticsearch/Document/DocumentCollection.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        $collection = new static();
+
+        foreach ($response->getDocuments() as $document) {
+            $collection->add($document);
+        }
+
+        return $collection;
+    }
+
+    public function add(DocumentInterface $document): void
```
Comment (dameert) [inline Elasticsearch/Document/DocumentCollection.php]: should be private I think

Comment (Davidmattei) [reply Elasticsearch/Document/DocumentCollection.php]: No I don't aggree, you should be able to create a collection yourself

File: Elasticsearch/Document/Document.php (hunk 1 above)
Comment (Davidmattei) [reply Elasticsearch/Document/Document.php]: Fixed

**Pull request merged**